front/components/assistant/AssistantBrowser.tsx

```diff
@@ -50,10 +51,15 @@ export function AssistantBrowser({
   const router = useRouter();
   const [assistantSearch, setAssistantSearch] = useState<string>("");
 
+  const whiteListedProviders = useMemo(() => {
+    return owner.whiteListedProviders ?? MODEL_PROVIDER_IDS;
```

Comment:
Let's add a function now that we've used this at least in 3 places?

Reply:
@spolu I tried creating the following function:
```typescript
export const getWhiteListedProvidersMemo = (
  workspace: WorkspaceType
) => {
  return useMemo(() => {
    return workspace.whiteListedProviders ?? MODEL_PROVIDER_IDS;
  }, [workspace.whiteListedProviders]);
};
```
But I kept on hitting the following error:
```
Error: FRONT_DATABASE_URI is required but not set
```
as if functions were not called in the right order (rule of hooks from what I gathered?). I change the memoization mechanism in one file, so I'm not sure we need a function anymore?